src/xls-reader/XlsReader.cs

```diff
@@ -144,29 +153,30 @@ public List<UcasCourseSubject> ReadCourseSubjects(string folder, IList<UcasCours
                         SubjectCode = row.GetCell(columnMap["SUBJECT_CODE"]).StringCellValue.Trim(),
                         YearCode = row.GetCell(columnMap["YEAR_CODE"]).StringCellValue.Trim()
                     };
+                    var skipMessage = $"Skipped invalid record in {CrseSubjectFilename} with inst_code {ucasCourseSubject.InstCode} and crse_code {ucasCourseSubject.CrseCode}.";
                     if (!courses.Any(c => c.InstCode == ucasCourseSubject.InstCode && c.CrseCode == ucasCourseSubject.CrseCode))
                     {
                         skipCount++;
-                        _logger.Warning($"  UcasCourseSubject skipped - invalid inst_code/crse_code combination: {ucasCourseSubject.InstCode}, {ucasCourseSubject.CrseCode}");
+                        _logger.Warning(skipMessage+"No valid record with this inst_code and crse_code found in {CrseFilename}");
                         continue;
                     }
                     if (!subjects.Any(c => c.SubjectCode == ucasCourseSubject.SubjectCode))
                     {
                         skipCount++;
-                        _logger.Warning($"  UcasCourseSubject skipped - invalid subject code: {ucasCourseSubject.SubjectCode} - inst/crse: {ucasCourseSubject.InstCode}, {ucasCourseSubject.CrseCode}");
+                        _logger.Warning(skipMessage+"No valid record with this subject_code found in {SubjectFilename}");
```

Comment:
Should there be a space between the sentences...

```suggestion
                        _logger.Warning(skipMessage+" No valid record with this subject_code found in {SubjectFilename}");
```